### ISSUE TITLE
Changing learning rate after loading state from a checkpoint

### ISSUE THREAD

**Issue by abhijitpal1247** [huggingface/accelerate#1591]
### System Info

```Shell
- `Accelerate` version: 0.19.0
- Platform: Linux-4.14.301-224.520.amzn2.x86_64-x86_64-with-glibc2.26
- Python version: 3.10.9
- Numpy version: 1.24.3
- PyTorch version (GPU?): 1.13.1+cu117 (True)
- System RAM: 62.13 GB
- GPU type: Tesla T4
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: NO
        - mixed_precision: fp16
        - use_cpu: False
        - num_processes: 1
        - machine_rank: 0
        - num_machines: 1
        - gpu_ids: 0
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [X] My own task or dataset (give details below)

### Reproduction

1. Finetune a dreambooth-lora model by using the provided script in the diffusers library
2. By making use of the resume from checkpoint functionality resume the finetuning from a checkpoint, while providing a different learning rate (lets say you had used 1e-4 in the previous finetuning iteration, now you want to continue with a lower learning rate 1e-5)
3. The learning rate remains the same (that is of previous iteration), how can we change the learning rate after resuming from a checkpoint?

### Expected behavior

Finetuning should continue with the new learning rate

**Comment by abhijitpal1247** [comment]
@sgugger 
I have tried this:
```
for opt in lr_scheduler.optimizers:
    for g in opt.param_groups:
        g['lr'] = new_lr
lr_scheduler.step()
```
but it isn't working

**Comment by abhijitpal1247** [comment]
```
for opt in lr_scheduler.optimizers:
      for g in opt.param_groups:
          g['lr'] = new_lr
  for g in optimizer:
      g['lr'] = new_lr
  for g in lr_scheduler.scheduler.optimizer:
      g['lr'] = new_lr
  optimizer.step()
  lr_scheduler.step()
```
its working now, thanks

**Comment by abhijitpal1247** [comment]
The issue got resolved with the above change. Closing the ticket.

**Comment by Luh1124** [comment]
You can modify the learning rate after resuming by changing the values of the base_lrs list in lr_schedule.
Please note that the lr_scheduler has been prepared by accelerate.prepare()

# modify learning rate scheduler to resume from the correct step
if args.learning_rate != lr_scheduler.scheduler.base_lrs[0]:
    lr_scheduler.scheduler.base_lrs = [args.learning_rate]
